### ISSUE TITLE
Adjusted headings from site-editor should get priority

### ISSUE THREAD

**Issue by merijnponzo** [WordPress/gutenberg#64498]
### Description

Whenever i try to set custom fontsizes to the headings in the site editor, there fontsizes are overwritten by 'inherit' values from other css files, like woo.css or even any css framework that contains a reset.scss etc where basicly headings always gets a font-size: inherit property

You can see that my h2 is getting a root value of 16px from the inherit property of the woocommerce reset. 

<img width="758" alt="Screenshot 2024-08-14 at 12 49 23" src="https://github.com/user-attachments/assets/783cad85-e263-43dd-ae16-69d4625448dd">

If i remove the inherit it's getting the correct fontsize:

<img width="745" alt="Screenshot 2024-08-14 at 12 48 22" src="https://github.com/user-attachments/assets/88afbf5f-3f2a-45b5-b06f-756ba1a139ed">

if the assigned headings could be prefixed with **:root :where(body)** the fontsize would be correct

### Step-by-step reproduction instructions

1. Go to your site-editor
Add a custom fontsize for for example an H2, add the h2 to your page and check the frontend
<img width="300" alt="Screenshot 2024-08-14 at 12 44 32" src="https://github.com/user-attachments/assets/09288c6b-2038-4d37-b971-a494c2c53c62">
2. Use any framework with a reset, like tailwind @base or woocommerce that loads woo.css 
My H2 is looking like this
<img width="475" alt="Screenshot 2024-08-14 at 12 44 52" src="https://github.com/user-attachments/assets/12a72fa1-d943-4a0c-ac83-07dd4f8e93a1">

### Environment info

wordpress 6.6.1

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by jorgefilipecosta** [comment]
I don't think it is something we can fix on core. Core is correctly setting the font size for h2 element. Using Woo or a css framework causes these styles to be overwritten, I guess the solution here is to not enqueue these reset stylesheets, to make sure global styles work as expected.
